### DISCUSSION TITLE
service 的计算方式是不是不太对

### DISCUSSION THREAD

**Discussion by idlist** [koishijs/koishi#464] [Q&A] [answered]
插件一多就开始报 service 太多了

把 maxListener 改成 0 的话，可以发现单 command 的插件也被算作一个 service，这是期望行为吗

**Comment by idlist** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/29368226/149902628-b29e7fcd-221f-497e-971e-02df31fe9ee1.png)

**Comment by shigma** [comment]
下个版本 (4.0.2) 将优化此问题。我调整了 service hook 的生成机制，service 钩子的数量将大幅减少，不至于轻易达到 64 个的限制。
